src/smc-webapp/project_page.cjsx

```diff
@@ -161,14 +162,32 @@ FreeProjectWarning = rclass ({name}) ->
         "#{name}" :
             free_warning_extra_shown : rtypes.bool
             free_warning_closed      : rtypes.bool
+        billing :
+            customer : rtypes.immutable  # similar to stripe_customer
 
     propTypes :
-        project_id : rtypes.string
+        project_id                           : rtypes.string
+        upgrades_you_can_use                 : rtypes.object
+        upgrades_you_applied_to_all_projects : rtypes.object
+        quota_params                         : rtypes.object.isRequired # from the schema
+        actions                              : rtypes.object.isRequired # projects actions
+
+    getInitialState: ->
+        state =
+            upgrading         : true
+            has_subbed        : false
+            state             : 'view'    # view --> edit --> saving --> view
+            title_text        : ''
+            description_text  : ''
+            error             : ''
+            create_button_hit : ''
```

Comment:
Why does this say `create_button_hit`? I'm guessing it was copy pasted from your previous code, but this name doesn't really make sense here.